src/components/Statistics.tsx

```diff
@@ -38,53 +38,83 @@ interface Props {
 const Statistics = ({ refresh }: Props) => {
   const { client } = useGlobalContext()
   const [hashrate, setHashrate] = useState('')
+  const [isHashrateLoading, setIsHashrateLoading] = useState(true)
   const [totalSupply, setTotalSupply] = useState<number>()
+  const [isTotalSupplyLoading, setIsTotalSupplyLoading] = useState(true)
   const [circulating, setCirculating] = useState<number>()
+  const [isCirculatingLoading, setIsCirculatingLoading] = useState(true)
   const [transactions, setTransactions] = useState<number>()
+  const [isTransactionsLoading, setIsTransactionsLoading] = useState(true)
   const [blocks, setBlocks] = useState<number>()
+  const [isBlocksLoading, setIsBlocksLoading] = useState(true)
   const [avgBlockTime, setAvgBlockTime] = useState<number>()
-  const [isLoading, setIsLoading] = useState(true)
+  const [isAvgBlockTimeLoading, setIsAvgBlockTimeLoading] = useState(true)
```

Comment:
This starts to be a lot of individual states!
I wonder if we could maybe group the loading states together, like (pseudo code, not tested):

```ts
type StatKey = 'hashrate' | 'totalSupply' | ...

type LoadingState = { [key in StatsKey]: boolean }

// Initiate all loadings to true
const [loadingState, setLoadingState] = useState<>({...})

// Little util fn to merge previous and new states
const updateLoadingState = (newState: Partial<LoadingState>) => {setLoadingState((prev) => {...prev, newState}}

updateLoadingState({ hashrate: false })
```

Reply:
The same technique could be used for storing and updating stats values.

```ts
type Stats = { [key in StatsKey]: number }

updateDataState({ hashrate: 42 })
```

Reply:
done

Reply:
I would just combine it all then into a `State`. 

```
type State = { [k in StatKey]: { value: number | string; isLoading: boolean } }; const [state, setState] = useState<State>(defaultState);
```

Reply:
I did that already